src/crafting.cpp

```diff
@@ -1070,9 +1079,11 @@ void player::complete_craft( item &craft, const tripoint &loc )
             first = false;
             // TODO: reconsider recipe memorization
             if( knows_recipe( &making ) ) {
-                add_msg( _( "You craft %s from memory." ), making.result_name() );
+                add_msg( _( "%s %s %s from memory." ), disp_name(), is_npc() ? _( "crafts" ) : _( "craft" ),
+                         making.result_name() );
             } else {
-                add_msg( _( "You craft %s using a book as a reference." ), making.result_name() );
+                add_msg( _( "%s %s %s using a book as a reference." ), disp_name(),
```

Comment:
```suggestion
                add_msg_player_or_npc( _( "You craft %s using a book as a reference." ), _( "<npcname> crafts %s using a book as a reference." ),  making.result_name() );
```

Your current construct is very difficult to translate as the translators won't know that "craft" / "crafts" belongs into this text and it's possible that the sentence structure is completely different for second person / third person.

Making two separate sentences makes it much simpler and also makes the code easier to read.

Same for the message above.